### PR TITLE
[Drop-In UI] Building Hightligth on Arrival

### PR DESCRIPTION
**Pull request by tomaszrybakiewicz**
Closes NAVAND-889

### Description
- Introduced `BuildingHighlightComponent`.
- Added `NavigationViewOptions.enableBuildingHighlightOnArrival` and `NavigationViewOptions.buildingHighlightOptions` 
- Updated `MapViewBinder`.
- Added `ComponentInstaller.buildingHighlight(...)`.

### Screenshots or Gifs

https://user-images.githubusercontent.com/2678039/204058831-035d560b-17cd-4855-92c4-83ac1d8136c0.mp4

### CODE REVIEW

**Comment by tomaszrybakiewicz** [comment]
@abhishek1508 I've enabled Building Hightligth by default. Please give it another go.

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/extensions/MapboxBuildingsApiExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.ui.maps.building.api.MapboxBuildingsApi
+import com.mapbox.navigation.ui.maps.building.model.BuildingError
+import com.mapbox.navigation.ui.maps.building.model.BuildingValue
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+@Throws(BuildingError::class)
+internal suspend fun MapboxBuildingsApi.queryBuildingToHighlight(
```
[the remark this answers is not in the document]
Comment (tomaszrybakiewicz) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/extensions/MapboxBuildingsApiExtensions.kt:12]: I was under impression that all our SDK API are callback based and any coroutine variants should be done in extensions. I can move them there if you would like, but what about all other extensions that convert callbacks to coroutines: MapboxNavigationExtensions, MapboxRouteLineApiExtensions, ResourceLoaderExtensions etc.? 
Shouldn't we move them as well?

**Comment by tomaszrybakiewicz** [comment]
@abhishek1508 I've added the missing Unit-tests and CHANGELOG entry. Please let me know if anything else needs to change.

**Review by abhishek1508 — APPROVED** [review]
Thank you for addressing the comments. LGTM

**Pull request merged**